service/storage_service.cc

```diff
@@ -5064,6 +5069,18 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(shar
            // Return an error since the command is from outdated leader
            co_return result;
         }
+
+        {
+            auto& state = _raft_topology_cmd_handler_state;
+            if (state.term != term) {
+                state.term = term;
```

Comment:
We're not using `state.term` for anything?

Reply:
`last_index` works only in the context of some `term`, if the `term` changes (coordinator failover took place) we just update the `term` and `last_index` without checking the `last_index`.

Reply:
Right - we use it because of the `else if`.

Reply:
Actually we never send uncommitted commands. For committed commands the term never changes. So it should be sufficient to track cmd_index, this will guarantee we never apply the command twice or from an outdated coordinator. What concerns me is that we can miss gaps in cmd_index, e..g due to a software bug we will skip some commands on the follower without noticing. next/prev state id approach guarantees this doesn't happen. but this is outside the scope of this patch.